src/Partition.cxx

```diff
@@ -225,6 +228,7 @@ Partition::OnIdleMonitor(unsigned mask) noexcept
 void
 Partition::OnGlobalEvent(unsigned mask) noexcept
 {
+	// TODO This doesn't seam to trigger when the random property of the queue is changed, this is needed for the pre caching
```

Comment:
A "sync with player" is only needed if the one next song changes; the player doesn't care for anything but the current and the next song. And the next song only if it has already finished decoding the current one; then it will start decoding the next one. That's why.